step-repositories/step-repositories-artifact/src/main/java/step/repositories/artifact/StepJarParser.java

```diff
@@ -18,108 +18,61 @@
  ******************************************************************************/
 package step.repositories.artifact;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 import step.automation.packages.AutomationPackageArchive;
 import step.automation.packages.AutomationPackageKeywordsAttributesApplier;
+import step.automation.packages.AutomationPackageReader;
 import step.automation.packages.AutomationPackageReadingException;
+import step.automation.packages.model.AutomationPackageContent;
 import step.automation.packages.model.AutomationPackageKeyword;
-import step.automation.packages.yaml.AutomationPackageDescriptorReader;
 import step.automation.packages.yaml.YamlAutomationPackageVersions;
-import step.automation.packages.yaml.model.AutomationPackageDescriptorYaml;
-import step.core.accessors.AbstractOrganizableObject;
-import step.core.dynamicbeans.DynamicValue;
 import step.core.plans.Plan;
 import step.core.scanner.AnnotationScanner;
 import step.functions.Function;
-import step.handlers.javahandler.Keyword;
-import step.junit.runner.StepClassParser;
-import step.junit.runner.StepClassParserResult;
-import step.junit.runners.annotations.Plans;
-import step.plans.nl.parser.PlanParser;
-import step.plugins.functions.types.CompositeFunctionUtils;
-import step.plugins.java.GeneralScriptFunction;
 import step.resources.ResourceManager;
 
 import java.io.File;
-import java.io.FileNotFoundException;
 import java.io.IOException;
-import java.io.InputStream;
-import java.lang.annotation.Annotation;
-import java.lang.reflect.Method;
-import java.net.JarURLConnection;
-import java.net.URL;
-import java.util.*;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public class StepJarParser {
 
-    private static final Logger logger = LoggerFactory.getLogger(StepJarParser.class);
-
-    private final StepClassParser stepClassParser;
-    private AutomationPackageDescriptorReader automationPackageDescriptorReader;
+    private final AutomationPackageReader automationPackageReader;
     private final AutomationPackageKeywordsAttributesApplier automationPackagesKeywordAttributesApplier;
 
     public StepJarParser() {
         this(null);
     }
 
     public StepJarParser(ResourceManager resourceManager) {
```

Comment:
Do we still need the StepJarParser?

Reply:
@jeromecomte 
`StepJarParser` is still used in `AbstractArtifactRepository`. When we are working with function package (for instance execute the function package via maven plugin), the `StepJarParser#getPlansForJar` is called.

I think, we can completely remove the `StepJarParser` when we stop supporting function packages and rewrite the `AbstractArtifactRepository`.

Or you want to move the remaining code from `StepJarParser` somewhere to the automation package module right now and delete the `StepJarParser` class?